evennia/utils/evmenu.py

```diff
@@ -64,10 +68,37 @@ def another_node(caller, input_string, **kwargs):
 The return values must be given in the above order, but each can be
 returned as None as well. If the options are returned as None, the
 menu is immediately exited and the default "look" command is called.
-
-    text (str, tuple or None): Text shown at this node. If a tuple, the
-        second element in the tuple is a help text to display at this
-        node when the user enters the menu help command there.
+    text (str, tuple, dict or None): Text shown at this node. If a dict
+        it can also contain various formatting instructions. In such a 
+        case the followng keys are already supported:        
+            - 'text' (str): The text of the node.
+            - 'help' (str): The help text of the node.
+            - 'footer' (str): Footer text for the node.
+            - 'format' (str): String dictating the layout of the node if
+                the default layout is not being used. It currently
+                supports two options:
```

Comment:
I like idea of more control over each node's formatting but I think this is a missed opportunity of flexibility. If having this 'format' key, it makes sense to also be able to supply a callable to it to perform the format. That would mean a user could achieve the effect it needs without hard-coding a specific format structure with a given name (which is the opposite of flexible). 

Maybe it would even make sense to be able to supply separate header- text- and footer-formatters when we're at it - falling back to the defaults if not given. Importing and setting the main text formatter to the `scroll` function would then replicate the scroll effect.

Reply:
I think this is actually three things.

The first is that perhaps the name should be something other than 'format'. Obviously most of the other options have formatting effects as well. The idea for this options was a to have a sort of large scale single word option that would lie over the 'gross results'. 

Using scroll as the example you can probably see how, other than the solid bars above and below the text that are part of the current default look, scroll is essentially the exact same menu with an 'effect' applied.

This is not a requirement of these functions, however. The whole idea behind them is that someone might want to be able to 'invert' certain menus for some reason (putting the options table above the text) so I didn't really want to call it just **effect** or **style**. I would suggest **frame** as long as we are willing to accept something like **suppress** or **invisible** as an option.

The second thing is that I wanted to move to a dictionary setup because we can extend it much more easily. We don't have to add separate header-, text-, and footer-formatters right now because if we can add them relatively easy later with little risk of breaking backwards compatibility due to complex parsing arguments (which allows us to avoid the 'perfect is the enemy of enemy of done' problem). 

The third thing is I don't think we should rely on replacing the main text formatter. The purpose of the (currently named) **format** option is to allow the main text format to be changed on a node-by-node basis. I originally wanted to suppress the top and bottom bars but only on specific nodes. Someone else might have a variety of layouts like **scroll** and want to switch between them (doing a scroll format when a person is in the 'magic' section of the menus and a format where the text is flanked by swords when they are in the 'fighting' section of the menus).